homeassistant/components/asuswrt/bridge.py

```diff
@@ -47,6 +59,42 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+async def migrate_legacy_protocols(
+    hass: HomeAssistant, conf: dict[str, Any]
+) -> dict[str, Any] | None:
+    """Migrate router protocol from legacy to http."""
+    if conf.get(CONF_SSH_KEY):
+        # migration temptative is not possible with ssh key
+        return None
+
+    if is_http(conf[CONF_PROTOCOL]):
+        return {}
+
+    sec_conf = {
+        k: v for k, v in conf.items() if k in [CONF_HOST, CONF_USERNAME, CONF_PASSWORD]
+    }
+    for protocol in HTTP_PROTOCOLS:
+        new_conf = {**sec_conf, CONF_PROTOCOL: protocol}
+        bridge = AsusWrtBridge.get_bridge(hass, new_conf)
+        try:
+            await bridge.async_connect()
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Error migrating host %s to protocol %s", new_conf[CONF_HOST], protocol
+            )
+            continue
+
+        await bridge.async_disconnect()
+        return new_conf
+
+    return {}
+
+
+def is_http(protocol: str) -> bool:
+    """Return true if protocol is http type."""
```

Comment:
```suggestion
    """Return true if protocol is HTTP type."""
```

TBH, I think this method is a bit too simple to warrant a method for... (but that is my personal opinion).

Reply:
Removed this method, used constant for comparison.